kernel/dosfns.c

```diff
@@ -967,13 +967,44 @@ COUNT DosGetExtFree(BYTE FAR * DriveString, struct xfreespace FAR * xfsp)
 
   if (cdsp->cdsFlags & CDSNETWDRV)
   {
-    if (remote_getfree(cdsp, rg) != SUCCESS)
-      return DE_INVLDDRV;
+    /* Try redirector extension */
+    if (remote_getfree_11a3(cdsp, rg) != SUCCESS)
+    {
+      /* Fallback */
+      if (remote_getfree(cdsp, rg) != SUCCESS)
+        return DE_INVLDDRV;
+
+      xfsp->xfs_clussize = rg[0];
+      xfsp->xfs_totalclusters = rg[1];
+      xfsp->xfs_secsize = rg[2];
+      xfsp->xfs_freeclusters = rg[3];
+    }
+    else /* Supports extension */
+    {
+      UDWORD total, avail;
+      UDWORD bps, spc;
+
+      bps = rg[4];
+      spc = 1;
+      total = ((rg[0] << 16UL) | rg[1]);
```

Comment:
When doing the FreeDOS version I got a compiler(ia16-gcc) warning here about left shift
~~~
 dosfns.c:921:23: error: left shift count >= width of type [-Werror=shift-count-overflow]
       total = ((rg[0] << 16UL) | rg[1]);
~~~

Casting it to
```c
    total = (((UDWORD)rg[0] << 16UL) | rg[1]);
```

Removed the warning. Do you think it's necessary here too?

Reply:
Cool thanks!